### ISSUE TITLE
Incorrect evaluated formula result for unbounded column

### ISSUE THREAD

**Issue by zi-l** [dgorissen/pycel#67]
#### What actually happened
The excel formula result of `SUM()/COUNTIFS` is incorrect for specific case:
```
=SUM(B15*Configuration!B4,B16*Configuration!B5,B17*Configuration!B6,B18*Configuration!B7,B19*Configuration!B8)
=COUNTIFS(Result!J:J,"Highest",Result!E:E,"fail")
```
![incorrect_evaluate_result](https://user-images.githubusercontent.com/25576713/64141354-4bffef80-ce3a-11e9-9034-7ccbba5f78d0.png)
formulas of `B15-B19`
![image](https://user-images.githubusercontent.com/25576713/64143989-20cecd80-ce45-11e9-8c6b-e041e0f779b7.png)

#### What was expected to happen
The excel formula result should be `-24` instead of `0`

#### Problem description
- Read excel through `ExcelCompiler`
- Calculate formula through `evaluate()`

#### Code Sample

```python
from pycel import ExcelCompiler
cp = ExcelCompiler(filename=PATH("data/TestData.xlsx"))
print("Report!B8: ", cp.evaluate('Report!B8'))
print("Report!B23: ", cp.evaluate('Report!B23'))
print("Report!B25: ", cp.evaluate('Report!B25'))
```

#### Environment
```
windows7 64bit
pycel:  1.0b16
python: 3.6.3
```

**Comment by stephenrauch** [comment]
It is going to be quite difficult to debug this without more information.  I built a quick test which replicated most of you showed here, and it worked perfectly.  Have you tried the `validate_calcs()` routine?

```
from pycel import ExcelCompiler
cp = ExcelCompiler(filename=PATH("data/TestData.xlsx"))
assert {} == cp.validate_calcs()
```

**Comment by zi-l** [comment]
@stephenrauch description is updated. I've tested with `openpyxl`, which causing this error.
formula of `B16`
```
=COUNTIFS(Result!J:J,"High",Result!E:E,"fail")
```
![normalwithoutopenpyxl](https://user-images.githubusercontent.com/25576713/64144423-1a415580-ce47-11e9-85f7-cd4001ce6578.png)
![abnormalwithopenpyxl](https://user-images.githubusercontent.com/25576713/64144426-1c0b1900-ce47-11e9-8dc6-11860d9b6c44.png)

**Comment by stephenrauch** [comment]
If you are going directly from openpyxl to pycel, you can get some unexpected results.  The basic problem is that openpyxl does not preserve the cached values that excel generally saves into the workbook.  So, if you read in a spreadsheet that openpyxl saved, it can confuse pycel.  

A few ways to try to work around this:

1) Use the `recalculate()` method which removes all of the cached values.
2) Recalculate in excel and then save before opening with pycel.
3) Use the method released in pycel 1.0b17 to take an openpyxl workbook directly when initializing ExcelCompiler via the `excel` keyword argument.

**Comment by zi-l** [comment]
@stephenrauch 
workaround `2` is the most unexpected solution. I have to finish the job with python, excluding open excel files with Microsoft Excel, OpenOffice or any other external program. 

`1` and `3` not work for my case. I've upgraded `pycel` to version `1.0.b17`, and tried to use `recalculate()` or pass `openpyxl` workbook instance to keyword argument `excel` of `ExcelCompiler` like:
```
# openpyxl
ex = excel(path=PATH("data/TestData.xlsx"))
ex.set_cell("Report", "B4", "2")
ex.set_cell("Report", "B2", "7")
ex.set_cell("Report", "B3", "9")
ex.save_excel()

# pycel
cp = ExcelCompiler(excel=ex.excelwb)    # excelwb is the instance of workbook
cp.recalculate()
print("Report!B8: ", cp.evaluate('Report!B8'))
print("Report!B16: ", cp.evaluate('Report!B16'))
print("Report!B25: ", cp.evaluate('Report!B25'))
```
ineffective
![image](https://user-images.githubusercontent.com/25576713/64150065-04895b80-ce5a-11e9-8c88-371397094fd4.png)

here's the error of `validate_calcs()`
```
Traceback (most recent call last):
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelformula.py", line 899, in eval_func
    excel_formula.compiled_lambda())
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelformula.py", line 920, in <lambda>
    ### Traceback will show this line if not loaded from a text file
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excellib.py", line 150, in countifs
    return len(handle_ifs(args))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1035, in handle_ifs
    for rng, criteria in zip(args[0::2], args[1::2])))
  File "C:\Users\biont.liu\AppData\Local\Programs\Python\Python36\lib\collections\__init__.py", line 535, in __init__
    self.update(*args, **kwds)
  File "C:\Users\biont.liu\AppData\Local\Programs\Python\Python36\lib\collections\__init__.py", line 622, in update
    _count_elements(self, iterable)
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1035, in <genexpr>
    for rng, criteria in zip(args[0::2], args[1::2])))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1118, in find_corresponding_index
    return tuple(find_corresponding_index_generator(rng, criteria))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1125, in find_corresponding_index_generator
    assert_list_like(rng)
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1137, in assert_list_like
    raise TypeError('Must be a list like: {}'.format(data))
TypeError: Must be a list like: Result!J1:J7
Eval: countifs(_R_("Result!J:J"), "Low", _R_("Result!E:E"), "fail")
Traceback (most recent call last):
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelformula.py", line 899, in eval_func
    excel_formula.compiled_lambda())
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelformula.py", line 920, in <lambda>
    ### Traceback will show this line if not loaded from a text file
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excellib.py", line 150, in countifs
    return len(handle_ifs(args))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1035, in handle_ifs
    for rng, criteria in zip(args[0::2], args[1::2])))
  File "C:\Users\biont.liu\AppData\Local\Programs\Python\Python36\lib\collections\__init__.py", line 535, in __init__
    self.update(*args, **kwds)
  File "C:\Users\biont.liu\AppData\Local\Programs\Python\Python36\lib\collections\__init__.py", line 622, in update
    _count_elements(self, iterable)
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1035, in <genexpr>
    for rng, criteria in zip(args[0::2], args[1::2])))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1118, in find_corresponding_index
    return tuple(find_corresponding_index_generator(rng, criteria))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1125, in find_corresponding_index_generator
    assert_list_like(rng)
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1137, in assert_list_like
    raise TypeError('Must be a list like: {}'.format(data))
TypeError: Must be a list like: Result!J1:J7
Eval: countifs(_R_("Result!J:J"), "Medium", _R_("Result!E:E"), "fail")
Traceback (most recent call last):
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelformula.py", line 899, in eval_func
    excel_formula.compiled_lambda())
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelformula.py", line 920, in <lambda>
    ### Traceback will show this line if not loaded from a text file
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excellib.py", line 150, in countifs
    return len(handle_ifs(args))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1035, in handle_ifs
    for rng, criteria in zip(args[0::2], args[1::2])))
  File "C:\Users\biont.liu\AppData\Local\Programs\Python\Python36\lib\collections\__init__.py", line 535, in __init__
    self.update(*args, **kwds)
  File "C:\Users\biont.liu\AppData\Local\Programs\Python\Python36\lib\collections\__init__.py", line 622, in update
    _count_elements(self, iterable)
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1035, in <genexpr>
    for rng, criteria in zip(args[0::2], args[1::2])))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1118, in find_corresponding_index
    return tuple(find_corresponding_index_generator(rng, criteria))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1125, in find_corresponding_index_generator
    assert_list_like(rng)
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1137, in assert_list_like
    raise TypeError('Must be a list like: {}'.format(data))
TypeError: Must be a list like: Result!J1:J7
Eval: countifs(_R_("Result!J:J"), "High", _R_("Result!E:E"), "fail")
Traceback (most recent call last):
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelformula.py", line 899, in eval_func
    excel_formula.compiled_lambda())
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelformula.py", line 920, in <lambda>
    ### Traceback will show this line if not loaded from a text file
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excellib.py", line 150, in countifs
    return len(handle_ifs(args))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1035, in handle_ifs
    for rng, criteria in zip(args[0::2], args[1::2])))
  File "C:\Users\biont.liu\AppData\Local\Programs\Python\Python36\lib\collections\__init__.py", line 535, in __init__
    self.update(*args, **kwds)
  File "C:\Users\biont.liu\AppData\Local\Programs\Python\Python36\lib\collections\__init__.py", line 622, in update
    _count_elements(self, iterable)
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1035, in <genexpr>
    for rng, criteria in zip(args[0::2], args[1::2])))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1118, in find_corresponding_index
    return tuple(find_corresponding_index_generator(rng, criteria))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1125, in find_corresponding_index_generator
    assert_list_like(rng)
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1137, in assert_list_like
    raise TypeError('Must be a list like: {}'.format(data))
TypeError: Must be a list like: Result!J1:J7
Eval: countifs(_R_("Result!J:J"), "Highest", _R_("Result!E:E"), "fail")
Traceback (most recent call last):
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelformula.py", line 899, in eval_func
    excel_formula.compiled_lambda())
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelformula.py", line 920, in <lambda>
    ### Traceback will show this line if not loaded from a text file
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excellib.py", line 150, in countifs
    return len(handle_ifs(args))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1035, in handle_ifs
    for rng, criteria in zip(args[0::2], args[1::2])))
  File "C:\Users\biont.liu\AppData\Local\Programs\Python\Python36\lib\collections\__init__.py", line 535, in __init__
    self.update(*args, **kwds)
  File "C:\Users\biont.liu\AppData\Local\Programs\Python\Python36\lib\collections\__init__.py", line 622, in update
    _count_elements(self, iterable)
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1035, in <genexpr>
    for rng, criteria in zip(args[0::2], args[1::2])))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1118, in find_corresponding_index
    return tuple(find_corresponding_index_generator(rng, criteria))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1125, in find_corresponding_index_generator
    assert_list_like(rng)
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1137, in assert_list_like
    raise TypeError('Must be a list like: {}'.format(data))
TypeError: Must be a list like: Result!J1:J7
Eval: countifs(_R_("Result!J:J"), "Highest", _R_("Result!E:E"), "fail")
Traceback (most recent call last):
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelformula.py", line 899, in eval_func
    excel_formula.compiled_lambda())
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelformula.py", line 920, in <lambda>
    ### Traceback will show this line if not loaded from a text file
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excellib.py", line 150, in countifs
    return len(handle_ifs(args))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1035, in handle_ifs
    for rng, criteria in zip(args[0::2], args[1::2])))
  File "C:\Users\biont.liu\AppData\Local\Programs\Python\Python36\lib\collections\__init__.py", line 535, in __init__
    self.update(*args, **kwds)
  File "C:\Users\biont.liu\AppData\Local\Programs\Python\Python36\lib\collections\__init__.py", line 622, in update
    _count_elements(self, iterable)
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1035, in <genexpr>
    for rng, criteria in zip(args[0::2], args[1::2])))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1118, in find_corresponding_index
    return tuple(find_corresponding_index_generator(rng, criteria))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1125, in find_corresponding_index_generator
    assert_list_like(rng)
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1137, in assert_list_like
    raise TypeError('Must be a list like: {}'.format(data))
TypeError: Must be a list like: Result!J1:J7
Eval: countifs(_R_("Result!J:J"), "Lowest", _R_("Result!E:E"), "fail")
Traceback (most recent call last):
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelformula.py", line 899, in eval_func
    excel_formula.compiled_lambda())
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelformula.py", line 920, in <lambda>
    ### Traceback will show this line if not loaded from a text file
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excellib.py", line 150, in countifs
    return len(handle_ifs(args))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1035, in handle_ifs
    for rng, criteria in zip(args[0::2], args[1::2])))
  File "C:\Users\biont.liu\AppData\Local\Programs\Python\Python36\lib\collections\__init__.py", line 535, in __init__
    self.update(*args, **kwds)
  File "C:\Users\biont.liu\AppData\Local\Programs\Python\Python36\lib\collections\__init__.py", line 622, in update
    _count_elements(self, iterable)
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1035, in <genexpr>
    for rng, criteria in zip(args[0::2], args[1::2])))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1118, in find_corresponding_index
    return tuple(find_corresponding_index_generator(rng, criteria))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1125, in find_corresponding_index_generator
    assert_list_like(rng)
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1137, in assert_list_like
    raise TypeError('Must be a list like: {}'.format(data))
TypeError: Must be a list like: Result!J1:J7
Eval: countifs(_R_("Result!J:J"), "Low", _R_("Result!E:E"), "fail")
Traceback (most recent call last):
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelformula.py", line 899, in eval_func
    excel_formula.compiled_lambda())
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelformula.py", line 920, in <lambda>
    ### Traceback will show this line if not loaded from a text file
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excellib.py", line 150, in countifs
    return len(handle_ifs(args))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1035, in handle_ifs
    for rng, criteria in zip(args[0::2], args[1::2])))
  File "C:\Users\biont.liu\AppData\Local\Programs\Python\Python36\lib\collections\__init__.py", line 535, in __init__
    self.update(*args, **kwds)
  File "C:\Users\biont.liu\AppData\Local\Programs\Python\Python36\lib\collections\__init__.py", line 622, in update
    _count_elements(self, iterable)
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1035, in <genexpr>
    for rng, criteria in zip(args[0::2], args[1::2])))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1118, in find_corresponding_index
    return tuple(find_corresponding_index_generator(rng, criteria))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1125, in find_corresponding_index_generator
    assert_list_like(rng)
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1137, in assert_list_like
    raise TypeError('Must be a list like: {}'.format(data))
TypeError: Must be a list like: Result!J1:J7
Eval: countifs(_R_("Result!J:J"), "Medium", _R_("Result!E:E"), "fail")
Traceback (most recent call last):
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelformula.py", line 899, in eval_func
    excel_formula.compiled_lambda())
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelformula.py", line 920, in <lambda>
    ### Traceback will show this line if not loaded from a text file
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excellib.py", line 150, in countifs
    return len(handle_ifs(args))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1035, in handle_ifs
    for rng, criteria in zip(args[0::2], args[1::2])))
  File "C:\Users\biont.liu\AppData\Local\Programs\Python\Python36\lib\collections\__init__.py", line 535, in __init__
    self.update(*args, **kwds)
  File "C:\Users\biont.liu\AppData\Local\Programs\Python\Python36\lib\collections\__init__.py", line 622, in update
    _count_elements(self, iterable)
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1035, in <genexpr>
    for rng, criteria in zip(args[0::2], args[1::2])))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1118, in find_corresponding_index
    return tuple(find_corresponding_index_generator(rng, criteria))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1125, in find_corresponding_index_generator
    assert_list_like(rng)
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1137, in assert_list_like
    raise TypeError('Must be a list like: {}'.format(data))
TypeError: Must be a list like: Result!J1:J7
Eval: countifs(_R_("Result!J:J"), "High", _R_("Result!E:E"), "fail")
Traceback (most recent call last):
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelformula.py", line 899, in eval_func
    excel_formula.compiled_lambda())
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelformula.py", line 920, in <lambda>
    ### Traceback will show this line if not loaded from a text file
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excellib.py", line 150, in countifs
    return len(handle_ifs(args))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1035, in handle_ifs
    for rng, criteria in zip(args[0::2], args[1::2])))
  File "C:\Users\biont.liu\AppData\Local\Programs\Python\Python36\lib\collections\__init__.py", line 535, in __init__
    self.update(*args, **kwds)
  File "C:\Users\biont.liu\AppData\Local\Programs\Python\Python36\lib\collections\__init__.py", line 622, in update
    _count_elements(self, iterable)
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1035, in <genexpr>
    for rng, criteria in zip(args[0::2], args[1::2])))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1118, in find_corresponding_index
    return tuple(find_corresponding_index_generator(rng, criteria))
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1125, in find_corresponding_index_generator
    assert_list_like(rng)
  File "C:\Users\biont.liu\PycharmProjects\ITF\venv\lib\site-packages\pycel\excelutil.py", line 1137, in assert_list_like
    raise TypeError('Must be a list like: {}'.format(data))
TypeError: Must be a list like: Result!J1:J7
Eval: countifs(_R_("Result!J:J"), "Highest", _R_("Result!E:E"), "fail")
Traceback (most recent call last):
  File "C:/Users/biont.liu/PycharmProjects/ITF/utils/celpy.py", line 72, in <module>
    assert {} == cp.validate_calcs()
AssertionError
0 formulas left to process

Process finished with exit code 1
```

**Comment by zi-l** [comment]
By the way, `B8` is always normal

**Comment by stephenrauch** [comment]
OK, this is way more information.  Not quite sure why it behaves this way, but the challenge is likely the entire (unbounded) column `"Result!J:J"` being passed in.  Support for these unbounded columns is new to the library.  I will try to take a look to see if I can reproduce this, and if I can then try to see if the unbounded range support can be extended for this use case.

In the mean time, if you can, I would try removing the unbounded column `J:J` and replacing with a bounded column `J1:J7`.  

```
TypeError: Must be a list like: Result!J1:J7
Eval: countifs(_R_("Result!J:J"), "Low", _R_("Result!E:E"), "fail")
```

**Comment by zi-l** [comment]
> OK, this is way more information. Not quite sure why it behaves this way, but the challenge is likely the entire (unbounded) column `"Result!J:J"` being passed in. Support for these unbounded columns is new to the library. I will try to take a look to see if I can reproduce this, and if I can then try to see if the unbounded range support can be extended for this use case.
> 
> In the mean time, if you can, I would try removing the unbounded column `J:J` and replacing with a bounded column `J1:J7`.
> 
> ```
> TypeError: Must be a list like: Result!J1:J7
> Eval: countifs(_R_("Result!J:J"), "Low", _R_("Result!E:E"), "fail")
> ```

I've tried to set formula to bounded column `J1:J7`, such as `B16`:
```
=COUNTIFS(Result!J1:J7,"High",Result!E1:E7,"fail")
```
It works fine now under this change. But I have to set the formula via python code, as the amount of `J:J`/`E:E` is not fixed and uncertain. Athough it's only a little more trouble to implement via coding, the bugfix will be much appreciated. 

Great thanks for your help, I'll have a deeper testing and comment here if any other problems.

**Comment by stephenrauch** [comment]
Thanks so much taking the time to report this issue.  I have released 1.0b18 to PyPI with code I think will resolve this issue.

**Comment by zi-l** [comment]
> 
> 
> Thanks so much taking the time to report this issue. I have released 1.0b18 to PyPI with code I think will resolve this issue.

verified as fixed with pycel v1.0b18
